server/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -178,20 +178,25 @@ protected AnalyzeResponse shardOperation(AnalyzeRequest request, ShardId shardId
     public static AnalyzeResponse analyze(AnalyzeRequest request, String field, Analyzer analyzer, IndexAnalyzers indexAnalyzers,
             AnalysisRegistry analysisRegistry, Environment environment, int maxTokenCount) throws IOException {
         boolean closeAnalyzer = false;
+        if (indexAnalyzers == null) {
+            Settings settings = Settings.builder()
+                .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+                .build();
+            IndexMetaData metaData = IndexMetaData.builder("_none_")
+                .settings(settings)
+                .numberOfShards(1)
+                .numberOfReplicas(1)
+                .build();
+            IndexSettings idxSettings = new IndexSettings(metaData, environment.settings());
+            indexAnalyzers = analysisRegistry.build(idxSettings);
```

Comment:
I'm not too familiar with the overhead that building the IndexAnalyzers for the "global" dummy index has. This way it seems it would get created new on every request, which hasn't happened before. Maybe there is a way of building this once (e.g. inside of AnalysisRegistry) and the reuse it? It looks like it should only contain pre-build analysis components that are not changing over time.

Reply:
Also I was wondering whether it is necessary to close the IndexAnalyzers after we are done if they are re-created every time here. This might not be necessary if they are created once inside the registry. I might be completely off with this as well, mostly asking for my own education.

Reply:
IndexAnalyzers will indeed need to be closed, and it's not entirely clear where that should happen.  I'm going to rethink the implementation here, this one is complicating things too much.